File: leetcode/range-sum-of-bst.java
//https://leetcode.com/problems/range-sum-of-bst/

class Solution {

    //Provided node class
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    public int rangeSumBST(TreeNode root, int low, int high) {
        return search(root, low, high, 0);
    }
    
    private int search(TreeNode curr, int low, int high, int sum){
        if(curr==null) return sum;
        
        if(curr.val>=low && curr.val<=high) sum+=curr.val;
    
        sum = search(curr.left, low, high, sum);
        sum = search(curr.right, low, high, sum);
        
        return sum;
    }
}

File: leetcode/range_sum_of_bst.py
# https://leetcode.com/problems/range-sum-of-bst


from typing import Optional


# Provided TreeNode class
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        range_sum: int = 0

        def traverse(node: TreeNode) -> None:
            nonlocal range_sum
            if node:
                if low <= node.val <= high:
                    range_sum += node.val
                traverse(node.left)
                traverse(node.right)

        traverse(root)
        return range_sum
